scheduling/allocator.rb

```diff
@@ -316,6 +318,11 @@ def calculate_score
       # penalty for ongoing vm provisionings on the host
       score += @candidate_host[:vm_provisioning_count] * 0.5
 
+      # prioritize AX102 for our premium CPU testers. Since we currently have
+      # a small number of AX102s in the fleet, decrease the score by 5
+      # instead of 0.5 to compensate the concurrent VM penalty.
```

Comment:
the concurrent VM penalty is only to avoid concurrent vm provisioning, not concurrently running vms. even for the performance use case, I'd assume we wouldn't want to provision many vms at the same time on a single host. we saw provisioning times spike for such cases.

Reply:
We currently have very few AX102 units in the fleet :facepalm: So, when a customer wants to test a premium CPU and requests runners more than that number in the same workflow, we can't provide premium runners for all jobs due to concurrent VM provisioning penalty. A few extra seconds in provisioning times aren't a big issue for them.

Reply:
just an idea: if we want to test AX102 with a preference for certain customers, we could do something like that:
`score += 0.5 * (@request.prioritize_performance_cpu ? -1 : 1) if @candidate_host[:total_cores] == 16`

otherwise, it's not unlikely that AX102 are already fully utilized when a tester comes along.